aws-logs-querydefinitions/src/test/java/software/amazon/logs/querydefinitions/BaseTests.java

```diff
@@ -0,0 +1,69 @@
+package software.amazon.logs.querydefinitions;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.services.cloudwatchlogs.model.InvalidParameterException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.ServiceUnavailableException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.doThrow;
+
+@ExtendWith(MockitoExtension.class)
+public class BaseTests {
```

Comment:
Should we test throttling behavior?

Reply:
I think its a bit janky/out of scope to do this in unit tests since the retry logic is first buried in the 'injectCredentialsAndCall' cfn method and then the sdk call methods, so it would be tough to mock a 5xx or whatever. If there is another way to test, I'm all for it though.